### PR TITLE
streamlink: Update to 6.11.0

### LINKED ISSUE #4056: Update streamlink to 6.11.0
### Please confirm there isn't already an existing package update request

- [X] I have searched through package update requests

### Name

streamlink

### Version

6.11.0

### Link to the release notes or announcement.

https://streamlink.github.io/changelog.html

### Reason for updating

I'm the maintainer of Streamlink, so I regularly check its packaging status on various Linux distros.

Solus' `streamlink` package has been outdated for a while now, almost half a year.
https://github.com/getsolus/packages/tree/373a8d39dfc152203b7f8c6b93f1a559fe8762b9/packages/s/streamlink

Applications like Streamlink which rely on unstable 3rd party APIs/websites require constant updates. There have been 8 new releases since Solus stopped updating the package.
https://streamlink.github.io/changelog.html

So I'm asking maintainers of Solus' `streamlink` package to have a look at packaging the most recent version, which currently is `6.11.0` (`7.0.0` coming up).

If the lack of package updates was caused by build issues of Streamlink's dependencies, then those should've been resolved ([`trio>=0.25`](https://github.com/python-trio/trio/releases/tag/v0.25.0) / [`trio-websocket`](https://github.com/python-trio/trio-websocket/commit/f5fd6d77db16a7b527d670c4045fa1d53e621c35)). I'll be glad to help if you run into any packaging issues.

I am not a user of Solus, I'm just interested in having my application updated to the latest version.

Thanks.

### PR DESCRIPTION
**Pull request by joebonrichie**
**Summary**
[Changelog](https://github.com/streamlink/streamlink/blob/6.11.0/CHANGELOG.md#streamlink-6110-2024-10-01)

Resolves #4056.

**Packaging Changes**
- Remove myself as maintainer
- Add monitoring info
- Enable testsuite

**Test Plan**

- Play a twitch stream in celluloid

**Checklist**

- [x] Package was built and tested against unstable

### CODE REVIEW

**Comment by sheepman4267** [comment]
I get a working stream in VLC, but not Celluloid. I've never used Celluloid before, though, so I may be doing something wrong.

**Comment by joebonrichie** [comment]
> I get a working stream in VLC, but not Celluloid. I've never used Celluloid before, though, so I may be doing something wrong.

you need the a different input option for celluloid e.g. fifo, the `-n` flag

File: packages/s/streamlink/package.yml [hunk 1]
```diff
@@ -23,6 +22,7 @@ builddeps  :
     - python-trio-websocket
     - python-typing-extensions
     - python-websocket-client
+    - python3-pycryptodome
 checkdeps  :
     - python-freezegun
     - python-pytest-asyncio
```
Comment (bastimeyer) [inline packages/s/streamlink/package.yml:28]: The `pytest-asyncio` dev-requirement has been removed in 6.11.0:
https://streamlink.github.io/changelog.html#streamlink-6-11-0-2024-10-01

Comment (joebonrichie) [reply packages/s/streamlink/package.yml]: Thanks, all the tests are passing now with the upstream changes

**Review by HarveyDevel — APPROVED** [review]
LGTM, thanks!

**Pull request merged**